include/libtorrent/torrent_handle.hpp

```diff
@@ -680,23 +680,51 @@ namespace aux {
 		static constexpr resume_data_flags_t save_info_dict = 1_bit;
 
-		// if nothing significant has changed in the torrent since the last
-		// time resume data was saved, fail this attempt. Significant changes
-		// primarily include more data having been downloaded, file or piece
-		// priorities, paused-state, transfer limit settings  having changed etc.
-		// Counters keeping statistics are notably *not* considered significant.
-		// Things like total_uploaded, total_downloaded, last_seen_complete,
-		// tracker scrape counters, are not considered significant enough
-		// changes for this flag to save the resume data.
-		// If the resume data doesn't need saving, a
-		// save_resume_data_failed_alert is posted with the error
-		// resume_data_not_modified.
+		// this flag has the same behavior as the combination of:
+		// if_counters_changed | if_download_progress | if_config_changed |
+		// if_state_changed | if_metadata_changed
 		static constexpr resume_data_flags_t only_if_modified = 2_bit;
 
-		// this flag amends the behavior of only_if_modified to include
-		// stats counter or scrape data updating to also count as an update, so
-		// it becomes more eager to save resume data.
-		static constexpr resume_data_flags_t save_counters = 3_bit;
+		// save resume data if any counters has changed since the last time
+		// resume data was saved. This includes upload/download counters, active
+		// time counters and scrape data. A torrent that is not paused will have
+		// its active time counters incremented continuously.
+		static constexpr resume_data_flags_t if_counters_changed = 3_bit;
+
+		// save the resume data if any blocks have been downloaded since the
+		// last time resume data was saved. This includes:
+		// * checking existing files on disk
+		// * downloading a block from a peer
+		// * files renamed or moved
+		static constexpr resume_data_flags_t if_download_progress = 4_bit;
+
+		// save the resume data if configuration options changed since last time
+		// the resume data was saved. This includes:
+		// * file- or piece priorities
+		// * upload- and download rate limits
+		// * change max-uploads (unchoke slots)
+		// * change max connection limit
+		// * enable/disable peer-exchange, local service discovery or DHT
+		// * enable/disable apply IP-filter
+		// * enable/disable auto-managed
+		// * enable/disable share-mode
+		// * enable/disable sequential-mode
+		static constexpr resume_data_flags_t if_config_changed = 5_bit;
+
+		// save the resume data if torrent state has changed since last time the
+		// resume data was saved. This includes:
+		// * upload mode
+		// * paused state
+		// * super-seeding
+		// * seed-mode
+		static constexpr resume_data_flags_t if_state_changed = 6_bit;
+
+		// save the resume data if any *metadata* changed since the last time
+		// resume data was saved. This includes:
+		// * add/remove web seeds
+		// * add/remove trackers
```

Comment:
It looks confusing. I would only refer to "info section" as metadata. Otherwise, wouldn't it be unexpected for a client who wants to save the resume data only after receiving the metadata for the magnet link, if instead it is saved when adding the tracker?

Reply:
it's not "instead", it's in addition to. "metadata" has traditionally meant the whole .torrent file. If one wants to save the .torrent file when it's received, trackers and web seeds will be part of it, and I would expect those to count too. But the water gets a bit muddy when considering "nodes" for instance (DHT nodes), which can also be included in torrent files.